src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -72,12 +76,54 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
             override fun visitArrayType(o: RsArrayType) = collectDiagnostics(holder, o)
             override fun visitVariantDiscriminant(o: RsVariantDiscriminant) = collectDiagnostics(holder, o)
             override fun visitPolybound(o: RsPolybound) = checkPolybound(holder, o)
+            override fun visitTryExpr(o: RsTryExpr) = checkTryExpr(holder, o)
             override fun visitTraitRef(o: RsTraitRef) = checkTraitRef(holder, o)
         }
 
         element.accept(visitor)
     }
 
+    private fun checkTryExpr(holder: AnnotationHolder, o: RsTryExpr) {
+        val items = o.knownItems
+        val lookup = ImplLookup(o.project, items)
+        val tryTrait = items.Try ?: return
+        val fromTrait = items.From ?: return
+
+        val tryExprTy = o.expr.type
+        if (tryExprTy is TyUnknown) return
+        val errorTy = findErrorType(tryExprTy, tryTrait, lookup)
+        if (errorTy == null) {
+            RsDiagnostic.TryTraitIsNotImplemented(o, tryExprTy).addToHolder(holder)
+            return
+        }
+
+        val returnTy = findParentFnOrLambdaRetTy(o)
+        val returnErrorTy = returnTy?.let { findErrorType(it, tryTrait, lookup) }
+        if (returnTy == null || returnErrorTy == null) {
```

Comment:
Also, if `returnTy` is `TyUnknown`, it should not add `TryTraitIsNotImplementedForReturnType` error.

Reply:
I'll fix it today

Reply:
Don't forget to add tests for such cases